### ISSUE TITLE
no-extra-parens should not raise an error when using parenthesis to wrap a multiline JSX

### ISSUE THREAD

**Issue by gajus** [eslint/eslint#3107]
Doing:

```js
return (
        <form>
            <input type='text' onChange={this.updateTask} />
            <button onClick={this.handleSubmit}>Add</button>
        </form>
    );
```

Will raise an error:

> error  Gratuitous parentheses around expression  no-extra-parens

I'd argue that parentheses in this case are necessary. Without them, I would have to move JSX inline with the `return` statement:

```js
return <form>
            <input type='text' onChange={this.updateTask} />
            <button onClick={this.handleSubmit}>Add</button>
        </form>;
```

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by gajus** [comment]
Having written down the requirement makes me question whether I am right. Parenthesis that are used only for style reasons are not necessary.

**Comment by michaelficarra** [comment]
Exactly. See previous issues. You can also style it as

```js
return 0,
  <...>;
```

**Comment by nzakas** [comment]
@gajus exactly. :)
